### DISCUSSION TITLE
How to copy a single file to multiple output locations?

### DISCUSSION THREAD

**Discussion by kevincox** [11ty/eleventy#3508] [Uncategorized]
I am trying to implement the [libravatar specification](https://wiki.libravatar.org/api/). Doing so requires making my avatar image available at a path specified by the MD5 and SHA256 hash of the email being checked.

My first attempt was this:

```typescript
for (let alg of ["md5", "sha256"]) {
	let hash = crypto.hash(alg, "kevincox@kevincox.ca");
	eleventyConfig.addPassthroughCopy({ "source/logo.png": `avatar/${hash}` });
}
```

However it seems that later calls overwrite earlier ones, so only the sha256 hash was being output.

Is there a way to copy a single file to multiple locations with eleventy?

**Comment by kevincox** [comment]
I did find an ugly workaround:

```typescript
for (let alg of ["md5", "sha256"]) {
	let hash = crypto.hash(alg, "kevincox@kevincox.ca");
	eleventyConfig.addPassthroughCopy(
		{ [`source/logo.png{,${hash}}`]: "avatar/" },
		{
			rename: () => `../${hash}`,
		},
	);
}
```

There are 2 major hacks here.

The first is using `{,${hash}}` in the input URL. This creates a glob that only matches the original file, but since the actual glob is unique it doesn't seem to get de-duplicated by eleventy.

The second hack is that because this is a "glob" rather than a file the file name would typically get appended to the output path. So I use the `rename` function to both remove the parent directory (which was `logo.png/`) and apply the actual file name I want.

Despite being incredibly ugly this seems to work and doesn't seem to be relying on any internal interfaces (other than maybe if the deduplication gets smarter in the future?)

**Reply by Ryuno-Ki** [reply]
https://github.com/11ty/eleventy/issues/326#issuecomment-2350947996 was using a copy after the build was finished.
Could work here since the input (the image) stays the same content-wise.

**Reply by kevincox** [reply]
That would work in this specific case as I don't really need these files in the dev server. But is also not a general solution.

**Reply by Ryuno-Ki** [reply]
The question was raised several times in the past, but often for templates or data-driven pages.
I haven't studied Eleventy's v3 current implementation to make recommendations on what would need to change.

Aside, you asked for „copy a single file to multiple output locations” so I guess there is your answer :)
You copy a single file to multiple output locations :nerd_face:

**Reply by kevincox** [reply]
Yes, but I was talking about using eleventy. Running regular nodejs commands after the build isn't really using the framework and you lose out on features like the dev server, ability to track lists of generated files, transforms and a handful of other features. I appreciate the answer, every workaround is helpful, but it seems like there is a missing feature here.

**Reply by Ryuno-Ki** [reply]
Let's wait for other ideas then.
